seisbench/generate/labeling.py

```diff
@@ -774,6 +776,208 @@ def __str__(self):
+            n = n / (np.std(n) + self.eps)
+
+        # Cutting noise to same length as x
+        if len(n) - len(x) < 0:
+            msg = (
+                f"The length of the data ({len(x)} samples) and the noise ({len(n)} samples) must either be the same "
+                f"or the shape of the noise must be larger than the shape of the data."
+            )
+            raise ValueError(msg)
+
+        if len(n) - len(x) > 0:
+            spoint = np.random.randint(low=0, high=len(n) - len(x))
+        else:
+            spoint = 0
+        n = n[spoint : spoint + len(x)]
+
+        # Scale earthquake (x) and noise (n) waveforms
+        x *= scale_earthquake
+        n *= scale_noise
+
+        # Add noise and signal to create noisy signal
+        noisy = x + n
+
+        # Normalize noisy, x and n by normalization factor of noisy to range [-1, 1]
+        if self.scaling_type == "peak":
+            norm_noisy = np.max(np.abs(noisy)) + self.eps
+        elif self.scaling_type == "std":
+            norm_noisy = np.std(noisy) + self.eps
+
+        # Scale earthquake, noise and noisy waveform with same scaling factor, that noisy is in range [-1, 1]
+        x = x / norm_noisy
+        n = n / norm_noisy
+        noisy = noisy / norm_noisy
+
+        # Perform STFT of x, n, and noisy
+        _, _, stft_x = stft(
+            x=x,
+            fs=self.sampling_rate or metadata["trace_sampling_rate_hz"],
+            nperseg=self.nperseg,
+            nfft=self.nfft,
+            **self.kwargs,
+        )
+        _, _, stft_n = stft(
+            x=n,
+            fs=self.sampling_rate or metadata["trace_sampling_rate_hz"],
+            nperseg=self.nperseg,
+            nfft=self.nfft,
+            **self.kwargs,
+        )
+        _, _, stft_noisy = stft(
+            x=noisy,
+            fs=self.sampling_rate or metadata["trace_sampling_rate_hz"],
+            nperseg=self.nperseg,
+            nfft=self.nfft,
+            **self.kwargs,
+        )
+
+        # Determine output masks
+        X = np.empty(
+            shape=(2, *stft_n.shape)
+        )  # Input, i.e. real and imag part of noisy
+
+        y = np.empty(
+            shape=(2, *stft_n.shape)
+        )  # Target, i.e. masks for signal and noise
+
+        # Normalize real and imaginary parts of STFT to range [-1, 1]
+        X[0, :, :] = stft_noisy.real / (np.max(np.abs(stft_noisy.real)) + self.eps)
+        X[1, :, :] = stft_noisy.imag / (np.max(np.abs(stft_noisy.imag)) + self.eps)
+
+        # Compute target masking function
+        y[0, :, :] = 1 / (1 + np.abs(stft_n) / (np.abs(stft_x) + self.eps))
+        y[1, :, :] = 1 - y[0, :, :]
+
+        # Replace nan values in X and y
+        X = np.nan_to_num(X)
+        y = np.nan_to_num(y)
+
+        # Update state_dicts for input and target
+        state_dict[self.key[0]] = (X, metadata)
+        state_dict[self.key[1]] = (y, metadata)
+
+    def label(self, X, metadata):
```

Comment:
I'm wondering if it might be an option to not subclass `SupervisedLabeller`. Semantically it seems reasonable, but on the other hand I think you're not actually using any functions of the superclass.